haystack/nodes/retriever/dense.py

```diff
@@ -1199,4 +1200,23 @@ def embed_documents(self, docs: List[Document]) -> List[np.ndarray]:
         :param docs: List of documents to embed
         :return: Embeddings, one per input document
         """
+        docs = self.linearize_tables(docs)  # linearize tables
         return self.embedding_encoder.embed_documents(docs)
+
+    def linearize_tables(self, docs: List[Document]) -> List[Document]:
+        """
+        Turns table documents into text documents by representing the table in csv format.
+        This allows us to use text embedding models for table retrieval.
+
+        :param docs: List of documents to linearize. If the document is not a table, it is returned as is.
+        :return: List of documents with linearized tables or original documents if they are not tables.
+        """
+        linearized_docs = []
+        for doc in docs:
+            if doc.content_type == "table":
+                doc = deepcopy(doc)
+                doc.content = doc.content.to_csv()  # type: ignore
```

Comment:
I think it would make sense to set `index=False` in `to_csv()`. Also, you can get rid of the `# type: ignore` by explicitly checking that `content` is of type `pd.DataFrame`:
```suggestion
                if isinstance(doc.content, pd.DataFrame):
                    doc.content = doc.content.to_csv(index=False)
```

Reply:
I have now added that, but mypy doesn't seem to get the hint.

Reply:
The problem here seems to be that you overwrite the `isinstance` check by creating a deepcopy of the document. The following should work I think:
```python
if doc.content_type == "table":
    doc = deepcopy(doc)
        if isinstance(doc.content, pd.DataFrame):
            doc.content = doc.content.to_csv(index=False)
```

Reply:
In that case, we should perhaps consider using an assert. This should still work with mypy according to [this](https://mypy.readthedocs.io/en/stable/common_issues.html) and we wouldn't have another if clause which would if false lead to unexpected behaviour instead of a crash.

It would then look like this:
``` py
if doc.content_type == "table":
    doc = deepcopy(doc)
    assert isinstance(doc.content, pd.DataFrame)
    doc.content = doc.content.to_csv(index=False)
```

Reply:
I think I would rather go with an else clause raising `HaystackError("Documents of type 'table' need to have a pd.DataFrame as content field')` instead if the assertion. This would provide a more meaningful error message to the user than a `AssertionError`.

Reply:
That makes sense. I have updated it accordingly.